rebased on upstream, bump iced version

doublerebel opened the pull request:
Hello,

Thanks for creating an iced gulp package.  I've rebased against the lastest gulp-coffee master, bumped iced to 1.7.x and added literate support so that all tests are passing.

Cheers,

Charles

Commit:
file.contents should be a buffer

ccheever committed:
coffee-script -> iced-coffee-script

ccheever committed:
Update README

dashed committed:
Add error handling info to README

With regards to https://github.com/wearefractal/gulp/issues/71

dashed committed:
Update README.md

dashed committed:
Add error handling info.

Based on https://github.com/wearefractal/gulp-coffee/pull/3#issuecomment-31239969

yocontra committed:
Merge pull request #3 from Dashed/patch-1

Add error handling info to README

yocontra committed:
drop support for under 0.9

Commit:
update deps

Commit:
fix copy pasta error

dashed committed:
Use es.through instead of es.map

See https://github.com/gulpjs/gulp/issues/75#issuecomment-31581317

Partial results for continue-on-error behaviour.

yocontra committed:
Merge pull request #4 from Dashed/patch-1

Use es.through instead of es.map

leny committed:
Adding some "real-code" based tests

yocontra committed:
Merge pull request #5 from leny/master

Adding some "real-code" based tests

leny committed:
Generate proper options to avoid gulp property pollution.

leny committed:
Adding more options for coffeescript compiler.

leny committed:
Remove unnecessary compiler option.

yocontra committed:
Merge pull request #6 from leny/master

Generate proper options to avoid gulp property pollution.

Commit:
fix #6

lettertwo committed:
Add basic source map support

The scenario in which this works correctly is one where
the source file, compiled file, and source map are all in the same
location. Additionally, since this plugin will stream two files
for every one input, subsequent commands may need to pass through
when the source map is streamed.

lettertwo committed:
Detect `literate` option by default

If the `literate` option is not defined, then the source file
path is examined to attempt to automatically detect the format.
This default behavior matches the expectation set by a similar
feature of the Coffeescript command when given a file path to parse.

yocontra committed:
Merge pull request #9 from lettertwo/detect-litcoffee

Detect `literate` option by default

lettertwo committed:
Create a new vinyl File for sourcemap

...instead of cloning the source and modifying it. As pointed out by
@Contra, cloning results in copying the original file buffer
unnecessarily.

lettertwo committed:
Remove compiled fixtures

Instead, fixtures compile along with the stream and the the results
compare against each other. This is a bit of futureproofing for
inevitable changes to coffeescript compiler output.

lettertwo committed:
Reorganize test suite to encapsulate repetative work

yocontra committed:
Merge pull request #8 from lettertwo/sourcemaps

Add basic source map support

Commit:
1.3

yocontra committed:
Merge pull request #10 from lettertwo/test-cleanup

Reorganize test suite to encapsulate repetative work

typicode committed:
Update coffee-script to 1.7.0

yocontra committed:
Merge pull request #12 from typicode/master

Update coffee-script to 1.7.0

Commit:
clean up, 1.4

zakuni committed:
update sourcemap syntax

update to //# syntax from deprecated //@

yocontra committed:
Merge pull request #13 from zakuni/sourcemap

update sourcemap syntax

Commit:
bump

yocontra committed:
fix readme ver, #20

yocontra committed:
update pkg ver, #20

hackerone committed:
added sourcemap option

yocontra committed:
Merge pull request #21 from hackerone/master

added source Root option

mattparlane committed:
Add header option (defaults to off)

yocontra committed:
Merge pull request #22 from mattparlane/master

Add header option (defaults to off)

kevva committed:
Add `data` var

Fixes #18.

yocontra committed:
Merge pull request #25 from kevva/patch-1

Add `data` var

Commit:
linting fixes and dep updates

Commit:
pin coffee version to 1.7 since they keep breaking things. fixes #28, closes #19

Commit:
adds compatibility for gulp-sourcemaps

Commit:
adds documentation for using gulp-sourcemaps

Commit:
apply the options configuration in the correct place, pass on the actual javascript content correctly if the coffeescript function returns an object

Commit:
switches to using vinyl-sourcemaps-apply

Commit:
fixes tests

Commit:
removes 0.9 and adds unstable tag for travis

Commit:
removes unstable tag :(

yocontra committed:
Merge pull request #33 from jessepollak/remove-node-0.9-from-travis

Removes 0.9 for travis

Commit:
Merge branch 'master' into gulp-sourcemap-compatible

yocontra committed:
Merge pull request #29 from jessepollak/gulp-sourcemap-compatible

Adds compatibility for gulp-sourcemaps

Commit:
2.0.0 - sourcemaps

Commit:
clean up options logic

Commit:
small docs cleanup

felixrabe committed:
package.json: Use CoffeeScript 1.7.1

Tested: `npm test` will pass.

To review the changes, see https://github.com/jashkenas/coffeescript/compare/1.7.0...1.7.1

yocontra committed:
Merge pull request #36 from felixrabe/patch-1

package.json: Use CoffeeScript 1.7.1

Commit:
2.0.1

svsool committed:
replace event-stream to through2 (Fix processing multiple files)

svsool committed:
remove debug strings from tests

svsool committed:
refactoring try/catch block

svsool committed:
pass file to next pipe if file empty

adamloving committed:
Fix sample code

extra semicolon, and missing gulp.dest

yocontra committed:
Merge pull request #41 from adamloving/patch-1

Fix sample code

yocontra committed:
Merge pull request #38 from svsool/master

replace event-stream with through2 (fix processing multiple files)

Commit:
2.1.0

floridoo committed:
fix source and destination path in source maps

yocontra committed:
Merge pull request #42 from floridoo/fix_sourcemap_paths

Fix source and destination path in source maps

Commit:
2.1.1

shinnn committed:
clean up .npmignore

yocontra committed:
Merge pull request #45 from shinnn/master

Update .npmignore

jackjennings committed:
fix renaming .coffee.md to .js

yocontra committed:
Merge pull request #48 from jackjennings/master

fix renaming .coffee.md to .js

Commit:
style changes and dep updates

Commit:
2.1.2

Commit:
update to coffee 1.8

doublerebel committed:
[doc] remaining coffee -> iced

doublerebel committed:
[fix] literate iced/coffee support.  requires ICS 1.7.1-f+.  all tests passing

doublerebel committed:
[dist] contributors for iced version

ccheever commented:
added you as a collaborator. there was a merge conflict and i won't have
time to resolve it for a few days, so you can just make all the changes you
want to the repo.
charlie

On Sun, Sep 28, 2014 at 12:36 PM, Charles Phillips <notifications@github.com

> wrote:
> 
> Hello,
> 
> Thanks for creating an iced gulp package. I've rebased against the lastest
> gulp-coffee master, bumped iced to 1.7.x and added literate support so that
> all tests are passing.
> 
> Cheers,
> 
> ## Charles
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/doublerebel/gulp-iced-coffee master
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/ccheever/gulp-iced-coffee/pull/1
> Commit Summary
> - file.contents should be a buffer
> - Add error handling info to README
> - Update README.md
> - Add error handling info.
> - Merge pull request #3 from Dashed/patch-1
> - drop support for under 0.9
> - update deps
> - fix copy pasta error
> - Use es.through instead of es.map
> - Merge pull request #4 from Dashed/patch-1
> - Adding some "real-code" based tests
> - Merge pull request #5 from leny/master
> - Generate proper options to avoid gulp property pollution.
> - Adding more options for coffeescript compiler.
> - Remove unnecessary compiler option.
> - Merge pull request #6 from leny/master
> - fix #6
> - Detect `literate` option by default
> - Merge pull request #9 from lettertwo/detect-litcoffee
> - Add basic source map support
> - Create a new vinyl File for sourcemap
> - Remove compiled fixtures
> - Merge pull request #8 from lettertwo/sourcemaps
> - 1.3
> - Reorganize test suite to encapsulate repetative work
> - Merge pull request #10 from lettertwo/test-cleanup
> - Update coffee-script to 1.7.0
> - Merge pull request #12 from typicode/master
> - clean up, 1.4
> - update sourcemap syntax
> - Merge pull request #13 from zakuni/sourcemap
> - bump
> - fix readme ver, #20
> - update pkg ver, #20
> - added sourcemap option
> - Merge pull request #21 from hackerone/master
> - Add header option (defaults to off)
> - Merge pull request #22 from mattparlane/master
> - Add `data` var
> - Merge pull request #25 from kevva/patch-1
> - linting fixes and dep updates
> - pin coffee version to 1.7 since they keep breaking things. fixes
>   #28, closes #19
> - adds compatibility for gulp-sourcemaps
> - adds documentation for using gulp-sourcemaps
> - apply the options configuration in the correct place, pass on the
>   actual javascript content correctly if the coffeescript function returns an
>   object
> - switches to using vinyl-sourcemaps-apply
> - fixes tests
> - removes 0.9 and adds unstable tag for travis
> - removes unstable tag :(
> - Merge pull request #33 from jessepollak/remove-node-0.9-from-travis
> - Merge branch 'master' into gulp-sourcemap-compatible
> - Merge pull request #29 from jessepollak/gulp-sourcemap-compatible
> - 2.0.0 - sourcemaps
> - clean up options logic
> - small docs cleanup
> - package.json: Use CoffeeScript 1.7.1
> - Merge pull request #36 from felixrabe/patch-1
> - 2.0.1
> - replace event-stream to through2 (Fix processing multiple files)
> - remove debug strings from tests
> - refactoring try/catch block
> - pass file to next pipe if file empty
> - Fix sample code
> - Merge pull request #41 from adamloving/patch-1
> - Merge pull request #38 from svsool/master
> - 2.1.0
> - fix source and destination path in source maps
> - Merge pull request #42 from floridoo/fix_sourcemap_paths
> - 2.1.1
> - clean up .npmignore
> - Merge pull request #45 from shinnn/master
> - fix renaming .coffee.md to .js
> - Merge pull request #48 from jackjennings/master
> - style changes and dep updates
> - 2.1.2
> - update to coffee 1.8
> - coffee-script -> iced-coffee-script
> - Update README
> - [doc] remaining coffee -> iced
> - [dist] bump iced to 1.7.x
> - [fix] literate iced/coffee support. requires ICS 1.7.1-f+. all tests
>   passing
> 
> File Changes
> - _M_ .npmignore
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-0 (4)
> - _M_ .travis.yml
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-1 (7)
> - _M_ README.md
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-2
>   (82)
> - _M_ index.js
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-3
>   (57)
> - _D_ lib/formatError.js
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-4 (6)
> - _M_ package.json
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-5
>   (27)
> - _A_ test/fixtures/grammar.iced
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-6
>   (621)
> - _A_ test/fixtures/journo.iced.md
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-7
>   (468)
> - _A_ test/fixtures/journo.liticed
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-8
>   (468)
> - _M_ test/main.js
>   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-9
>   (273)
> 
> Patch Links:
> - https://github.com/ccheever/gulp-iced-coffee/pull/1.patch
> - https://github.com/ccheever/gulp-iced-coffee/pull/1.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ccheever/gulp-iced-coffee/pull/1.

ccheever commented:
thanks for the patch!

On Sun, Sep 28, 2014 at 10:49 PM, Charlie Cheever ccheever@gmail.com
wrote:

> added you as a collaborator. there was a merge conflict and i won't have
> time to resolve it for a few days, so you can just make all the changes you
> want to the repo.
> charlie
> 
> On Sun, Sep 28, 2014 at 12:36 PM, Charles Phillips <
> notifications@github.com> wrote:
> 
> > Hello,
> > 
> > Thanks for creating an iced gulp package. I've rebased against the
> > lastest gulp-coffee master, bumped iced to 1.7.x and added literate support
> > so that all tests are passing.
> > 
> > Cheers,
> > 
> > ## Charles
> > 
> > You can merge this Pull Request by running
> > 
> >   git pull https://github.com/doublerebel/gulp-iced-coffee master
> > 
> > Or view, comment on, or merge it at:
> > 
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1
> > Commit Summary
> > - file.contents should be a buffer
> > - Add error handling info to README
> > - Update README.md
> > - Add error handling info.
> > - Merge pull request #3 from Dashed/patch-1
> > - drop support for under 0.9
> > - update deps
> > - fix copy pasta error
> > - Use es.through instead of es.map
> > - Merge pull request #4 from Dashed/patch-1
> > - Adding some "real-code" based tests
> > - Merge pull request #5 from leny/master
> > - Generate proper options to avoid gulp property pollution.
> > - Adding more options for coffeescript compiler.
> > - Remove unnecessary compiler option.
> > - Merge pull request #6 from leny/master
> > - fix #6
> > - Detect `literate` option by default
> > - Merge pull request #9 from lettertwo/detect-litcoffee
> > - Add basic source map support
> > - Create a new vinyl File for sourcemap
> > - Remove compiled fixtures
> > - Merge pull request #8 from lettertwo/sourcemaps
> > - 1.3
> > - Reorganize test suite to encapsulate repetative work
> > - Merge pull request #10 from lettertwo/test-cleanup
> > - Update coffee-script to 1.7.0
> > - Merge pull request #12 from typicode/master
> > - clean up, 1.4
> > - update sourcemap syntax
> > - Merge pull request #13 from zakuni/sourcemap
> > - bump
> > - fix readme ver, #20
> > - update pkg ver, #20
> > - added sourcemap option
> > - Merge pull request #21 from hackerone/master
> > - Add header option (defaults to off)
> > - Merge pull request #22 from mattparlane/master
> > - Add `data` var
> > - Merge pull request #25 from kevva/patch-1
> > - linting fixes and dep updates
> > - pin coffee version to 1.7 since they keep breaking things. fixes
> >   #28, closes #19
> > - adds compatibility for gulp-sourcemaps
> > - adds documentation for using gulp-sourcemaps
> > - apply the options configuration in the correct place, pass on the
> >   actual javascript content correctly if the coffeescript function returns an
> >   object
> > - switches to using vinyl-sourcemaps-apply
> > - fixes tests
> > - removes 0.9 and adds unstable tag for travis
> > - removes unstable tag :(
> > - Merge pull request #33 from jessepollak/remove-node-0.9-from-travis
> > - Merge branch 'master' into gulp-sourcemap-compatible
> > - Merge pull request #29 from jessepollak/gulp-sourcemap-compatible
> > - 2.0.0 - sourcemaps
> > - clean up options logic
> > - small docs cleanup
> > - package.json: Use CoffeeScript 1.7.1
> > - Merge pull request #36 from felixrabe/patch-1
> > - 2.0.1
> > - replace event-stream to through2 (Fix processing multiple files)
> > - remove debug strings from tests
> > - refactoring try/catch block
> > - pass file to next pipe if file empty
> > - Fix sample code
> > - Merge pull request #41 from adamloving/patch-1
> > - Merge pull request #38 from svsool/master
> > - 2.1.0
> > - fix source and destination path in source maps
> > - Merge pull request #42 from floridoo/fix_sourcemap_paths
> > - 2.1.1
> > - clean up .npmignore
> > - Merge pull request #45 from shinnn/master
> > - fix renaming .coffee.md to .js
> > - Merge pull request #48 from jackjennings/master
> > - style changes and dep updates
> > - 2.1.2
> > - update to coffee 1.8
> > - coffee-script -> iced-coffee-script
> > - Update README
> > - [doc] remaining coffee -> iced
> > - [dist] bump iced to 1.7.x
> > - [fix] literate iced/coffee support. requires ICS 1.7.1-f+. all
> >   tests passing
> > 
> > File Changes
> > - _M_ .npmignore
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-0
> >   (4)
> > - _M_ .travis.yml
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-1
> >   (7)
> > - _M_ README.md
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-2
> >   (82)
> > - _M_ index.js
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-3
> >   (57)
> > - _D_ lib/formatError.js
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-4
> >   (6)
> > - _M_ package.json
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-5
> >   (27)
> > - _A_ test/fixtures/grammar.iced
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-6
> >   (621)
> > - _A_ test/fixtures/journo.iced.md
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-7
> >   (468)
> > - _A_ test/fixtures/journo.liticed
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-8
> >   (468)
> > - _M_ test/main.js
> >   https://github.com/ccheever/gulp-iced-coffee/pull/1/files#diff-9
> >   (273)
> > 
> > Patch Links:
> > - https://github.com/ccheever/gulp-iced-coffee/pull/1.patch
> > - https://github.com/ccheever/gulp-iced-coffee/pull/1.diff
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/ccheever/gulp-iced-coffee/pull/1.

Commit:
update mocha

doublerebel committed:
[doc] 2014, 2015 Copyright notice

stephenlacy committed:
update deps

stephenlacy committed:
update .ignore files

stephenlacy committed:
lint and update

stephenlacy committed:
switch to docker travis

Commit:
Merge pull request #55 from stevelacy/master

Update deps + lint

Commit:
2.2.1

alubbe committed:
update coffee to 1.9

alubbe committed:
2.3.0

alubbe committed:
fixed tests not working on windows

Commit:
Merge pull request #57 from alubbe/master

2.3.0 with coffee 1.9.0

Commit:
2.3.1

pdehaan committed:
Update license attribute

specifying the type and URL is deprecated:

https://docs.npmjs.com/files/package.json#license
http://npm1k.org/

Commit:
Merge pull request #63 from pdehaan/patch-1

Update license attribute

stephenlacy committed:
add node v4 to travis

Commit:
Merge pull request #65 from stevelacy/master

add node v4 to travis

KCreate committed:
added semicolon

Commit:
Merge pull request #71 from KCreate/patch-1

added semicolon

joaumg committed:
fix(travisbadge & typo): README.md

Commit:
Merge pull request #72 from joaumg-forks/master

fix(travisbadge & typo): README.md

brandonson committed:
Update coffeescript to 1.10.0

Commit:
Merge pull request #73 from brandonson/master

Update coffeescript to 1.10.0

Commit:
2.3.2

Commit:
Merge pull request #75 from pmb0/master

2.3.2

doublerebel committed:
[doc] more coffee -> iced

doublerebel committed:
[test] iced badge; Node 4.x, 5.x

doublerebel commented:
Thanks again.  Finally brought this current and merged to master.  @ccheever are you still the owner of this npm package?
